### DISCUSSION TITLE
Error launching Mirth

### DISCUSSION THREAD

**Discussion by edsalvador** [nextgenhealthcare/connect#5938] [General]
I am using Mirth 4.1.1.
I have a site that we are trying to connect with that requires SSL cert.
I added the cert to C:\Program Files\Mirth Connect\appdata\keystore.jks.
However, when I run the Launcher I get the following Error:
Unable to retrieve payload from HTTP request. URI: https://localhost:8443

**Comment by litemikx** [comment]
I usually get that error when the service hasn't fully started yet, or if you just restarted it. For the SSL cert, I'm not sure if I'm wrong but from what I do, the SSL cert should be added in the Java CA keystore of your Java runtime environment, either on the bundle Mirth or the system. I wrote a tutorial here: https://thisdevblogs.wordpress.com/2022/09/02/mirth-nextgen-connect-bits-5-adding-ssl-certificate-to-resolve-https-sender-issues/ 

Downside here is you can't monitor the expiry of the cert.

**Reply by pacmano1** [reply]
Well, other downsides to your article is it shares the solution using all MS (or deskop) tools, no command line instructions.  Also, in many cases folks would need to transfer keystores to and from servers if using KeyStore Explorer as THE tool the modify keystores.

Busting your chops moslty of course, it is a helpful article of course!

**Reply by litemikx** [reply]
Yeah, this was a bit rough, this was way back I think 5 years ago. I haven't use this in a production client because we have Network peeps now. The more effective way I see is to just have the network route the outbound request to a local port and route it to the https endpoint, eliminating the need for Mirth to handle the SSL. Mirth will just need to call the localhost port. Not sure if I explained correctly though :D

**Reply by pacmano1** [reply]
Sure, tools like haproxy and stunnel can do that (offload TLS).

**Comment by edsalvador** [comment]
Here are the steps that took place:
 - We created a CSR and sent to the site
 - The site sent back .cer and .p7b
 - We created a .pfx
 - I imported the .pfx into the Mirth keystore.jks (I used portecle)
 - Mirth will not launch with the Error

**Comment by edsalvador** [comment]
Any Suggestions?

**Comment by edsalvador** [comment]
This issue has been resolved

**Reply by pacmano1** [reply]
It would be helpful to other to share your fix.

**Comment by edsalvador** [comment]
I deleted the cert from keystore.jks and then added again.
It worked.
